### PR TITLE
Allow user define specific path for output resource

### PR DESCRIPTION
**Pull request by vincent-pli**
# Changes
Use TargetPath in resource definition to point source path of output resource. 
#841 
# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: pkg/reconciler/v1alpha1/taskrun/resources/output_resource.go [hunk 1]
```diff
@@ -97,7 +97,11 @@ func AddOutputResources(
 		// To build copy step it needs source path(which is targetpath of input resourcemap) from task input source
 		sourcePath := inputResourceMap[boundResource.Name]
 		if sourcePath == "" {
-			sourcePath = filepath.Join(outputDir, boundResource.Name)
+			if output.TargetPath == "" {
+				sourcePath = filepath.Join(outputDir, boundResource.Name)
+			} else {
+				sourcePath = output.TargetPath
+			}
```
Comment (bobcatfish) [inline pkg/reconciler/v1alpha1/taskrun/resources/output_resource.go]: could you add some unit test coverage for this?

Comment (vincent-pli) [reply pkg/reconciler/v1alpha1/taskrun/resources/output_resource.go]: Sure!

**Comment by vincent-pli** [comment]
@bobcatfish 
I added the UT and related document, please take a look, thanks.

**Comment by bobcatfish** [comment]
Thanks @vincent-pli !!

/lgtm
/approve
/meow space

**Pull request merged**